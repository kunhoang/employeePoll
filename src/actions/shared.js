import { receiveUsers } from "./users";
import { receiveQuestions } from "./questions";
import { getInitialData } from "../util/api";

export function handleInitialData() {
  return async (dispatch) => {
    const { users, questions } = await getInitialData();
      dispatch(receiveUsers(users));
      dispatch(receiveQuestions(questions));
  };
}
