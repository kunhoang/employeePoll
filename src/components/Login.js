import "../assest/login.css";
import image from "../image/image.jpg";
import { useNavigate, Routes, Route } from "react-router-dom";
import { connect } from "react-redux";
import { useState } from "react";
import { handleLogin } from "../actions/authedUser";
const Login = ({ dispatch, loggedIn }) => {
  const [username, setUsername] = useState("sarahedo");
  const [password, setPassword] = useState("password123");

  const handleUsername = (e) => {
    setUsername(e.target.value);
  };

  const handlePassword = (e) => {
    setPassword(e.target.value);
  };
  const navigate = useNavigate();
  // if (loggedIn) {
  //   navigate("/");
  // }

  const handleSubmit = () => {
    dispatch(handleLogin(username, password));
    setUsername("");
    setPassword("");
    navigate("/");
  };

  return (
    <div className="main" data-testid="login">
      <div className="title">Employee Polls</div>
      <div className="logo">
        <img src={image} className="img" />
      </div>
      <div className="form">
        <div className="form-title">Login</div>
        <div className="form-user">
          <span className="form-user-text">User</span>
          <input
            type="text"
            className="text-user"
            value={username}
            onChange={handleUsername}
            data-testid = "username"
          />
        </div>
        <div className="form-password">
          <span className="form-password-text">Password</span>
          <input
            type="password"
            className="text-password"
            value={password}
            onChange={handlePassword}
            data-testid = "password"
          />
        </div>
        <button className="btn-submit" onClick={handleSubmit} data-testid = "submit">
          Submit
        </button>
      </div>
    </div>
  );
};

const mapStateToProps = ({ authedUser }) => ({
  loggedIn: !!authedUser,
});

export default connect(mapStateToProps)(Login);
