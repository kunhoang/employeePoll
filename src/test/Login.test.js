import {fireEvent, render} from "@testing-library/react";
import {Provider} from "react-redux";
import {store} from "../store";
import {BrowserRouter} from "react-router-dom";
import React from "react";
import Login from "../components/Login";
import {handleInitialData} from "../actions/shared";

describe("Login", () => {
    it("Check component", () => {
        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <Login/>
                </BrowserRouter>
            </Provider>
        );
        expect(component).toBeDefined();
        expect(component).toMatchSnapshot();
    });

    it('Check handle login', async () => {
        await store.dispatch(handleInitialData());

        const wrapper = render(
            <Provider store={store}>
                <BrowserRouter>
                    <Login/>
                </BrowserRouter>
            </Provider>
        );

        const dataHeading = wrapper.getByTestId("login");
        const dataUsername = wrapper.getByTestId("username");
        const dataPassword = wrapper.getByTestId("password");
        const dataSubmitBtn = wrapper.getByTestId("submit");
        expect(dataHeading).toBeInTheDocument();
        expect(dataUsername).toBeInTheDocument();
        expect(dataPassword).toBeInTheDocument();
        expect(dataSubmitBtn).toBeInTheDocument();

        fireEvent.change(dataUsername, {target: {value: 'tylermcginnis'}});
        fireEvent.change(dataPassword, {target: {value: '123'}});
        expect(dataUsername.value).toBe("tylermcginnis");
        expect(dataPassword.value).toBe("123");
        fireEvent.click(dataSubmitBtn);
        expect(dataHeading).toBeInTheDocument();
        expect(dataUsername.value).toBe("");
        expect(dataPassword.value).toBe("");
    });
});