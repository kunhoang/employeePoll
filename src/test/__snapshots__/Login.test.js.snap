// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Check component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main"
        data-testid="login"
      >
        <div
          class="title"
        >
          Employee Polls
        </div>
        <div
          class="logo"
        >
          <img
            class="img"
            src="image.jpg"
          />
        </div>
        <div
          class="form"
        >
          <div
            class="form-title"
          >
            Login
          </div>
          <div
            class="form-user"
          >
            <span
              class="form-user-text"
            >
              User
            </span>
            <input
              class="text-user"
              data-testid="username"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-password"
          >
            <span
              class="form-password-text"
            >
              Password
            </span>
            <input
              class="text-password"
              data-testid="password"
              type="password"
              value=""
            />
          </div>
          <button
            class="btn-submit"
            data-testid="submit"
          >
            Submit
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main"
      data-testid="login"
    >
      <div
        class="title"
      >
        Employee Polls
      </div>
      <div
        class="logo"
      >
        <img
          class="img"
          src="image.jpg"
        />
      </div>
      <div
        class="form"
      >
        <div
          class="form-title"
        >
          Login
        </div>
        <div
          class="form-user"
        >
          <span
            class="form-user-text"
          >
            User
          </span>
          <input
            class="text-user"
            data-testid="username"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-password"
        >
          <span
            class="form-password-text"
          >
            Password
          </span>
          <input
            class="text-password"
            data-testid="password"
            type="password"
            value=""
          />
        </div>
        <button
          class="btn-submit"
          data-testid="submit"
        >
          Submit
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
